src/main/java/net/neoforged/neoforge/transfer/handlers/templates/resource/IResourceStorageData.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer.handlers.templates.resource;
+
+import java.util.Iterator;
+import net.neoforged.neoforge.transfer.resources.IResource;
+import net.neoforged.neoforge.transfer.resources.IResourceStack;
+import org.jetbrains.annotations.Contract;
+
+public interface IResourceStorageData<R extends IResource> extends Iterable<IResourceStack<R>> {
```

Comment:
I had another look, and this seems like it could just be replaced by storing in a list of stacks.